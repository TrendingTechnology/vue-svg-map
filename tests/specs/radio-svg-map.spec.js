import Vue from 'vue'
import { mount } from '@vue/test-utils'
import map from '../mocks/fake-map'
import SvgMap from '../../src/components/svg-map'
import RadioSvgMap from '../../src/components/radio-svg-map'

describe('RadioSvgMap component', () => {
	const value = 'id1'
	let wrapper
	let svgMap
	let locations

	beforeEach(() => {
		wrapper = mount(RadioSvgMap, {
			propsData: { map, value },
		})
		svgMap = wrapper.find(SvgMap)
		locations = svgMap.findAll('path')
	})

	it('selects location of inital value property', () => {
		expect(locations.at(0).attributes('aria-checked')).toBeFalsy()
		expect(locations.at(1).attributes('aria-checked')).toBeTruthy()
		expect(locations.at(2).attributes('aria-checked')).toBeFalsy()
	})

	it('selects location of new value property', async () => {
		wrapper.setProps({ value: 'id0' })
		await Vue.nextTick()

		expect(locations.at(0).attributes('aria-checked')).toBeTruthy()
		expect(locations.at(1).attributes('aria-checked')).toBeFalsy()
		expect(locations.at(2).attributes('aria-checked')).toBeFalsy()
	})

	it('sets only selected location as focusable', () => {
		expect(locations.at(0).attributes('tabindex')).toBe('-1')
		expect(locations.at(1).attributes('tabindex')).toBe('0')
		expect(locations.at(2).attributes('tabindex')).toBe('-1')
	})

	it('sets only first location as focusable when no selected location', async () => {
		wrapper.setProps({ value: null })
		await Vue.nextTick()

		expect(locations.at(0).attributes('tabindex')).toBe('0')
		expect(locations.at(1).attributes('tabindex')).toBe('-1')
		expect(locations.at(2).attributes('tabindex')).toBe('-1')
	})

	it('emits new selected location when clicking on unselected location', () => {
		locations.at(2).trigger('click')

		expect(wrapper.emitted().change[0][0]).toEqual('id2')
	})

	it('emits nothing when clicking on selected location', () => {
		locations.at(1).trigger('click')

		expect(wrapper.emitted().change).toBeUndefined()
	})

	it('emits new selected location when pressing spacebar on unselected location', () => {
		locations.at(2).trigger('keydown.space')

		expect(wrapper.emitted().change[0][0]).toEqual('id2')
	})

	it('emits nothing when pressing spacebar on selected location', () => {
		locations.at(1).trigger('keydown.space')

		expect(wrapper.emitted().change).toBeUndefined()
	})

	it('emits next location when pressing down or right arrow', () => {
		locations.at(1).trigger('keydown.down')

		expect(wrapper.emitted().change[0][0]).toEqual('id2')
	})

	it('emits first location when pressing down or right arrow on last location', async () => {
		wrapper.setProps({ value: 'id2' })
		await Vue.nextTick()
		locations.at(2).trigger('keydown.right')

		expect(wrapper.emitted().change[0][0]).toEqual('id0')
	})

	it('emits previous location when pressing up or left arrow', () => {
		locations.at(1).trigger('keydown.up')

		expect(wrapper.emitted().change[0][0]).toEqual('id0')
	})

	it('emits last location when pressing up or left arrow on first location', async () => {
		wrapper.setProps({ value: 'id0' })
		await Vue.nextTick()
		locations.at(0).trigger('keydown.left')

		expect(wrapper.emitted().change[0][0]).toEqual('id2')
	})

	it('renders map with radio button attributes', () => {
		expect(wrapper.element).toMatchSnapshot()
	})
})
