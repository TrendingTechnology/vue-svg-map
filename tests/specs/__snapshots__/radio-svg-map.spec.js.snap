// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioSvgMap component renders map with radio button attributes 1`] = `
<svg
  aria-label="label"
  class="svg-map"
  map="[object Object]"
  role="radiogroup"
  viewBox="viewBox"
  xmlns="http://www.w3.org/2000/svg"
>
   
  <path
    aria-label="name0"
    class="svg-map__location"
    d="path0"
    id="id0"
    name="name0"
    role="radio"
    tabindex="-1"
  />
  <path
    aria-checked="true"
    aria-label="name1"
    class="svg-map__location"
    d="path1"
    id="id1"
    name="name1"
    role="radio"
    tabindex="0"
  />
  <path
    aria-label="name2"
    class="svg-map__location"
    d="path2"
    id="id2"
    name="name2"
    role="radio"
    tabindex="-1"
  />
   
</svg>
`;
